packages/headless/src/components/TemplateLoader.tsx

```diff
@@ -22,11 +21,14 @@ export interface WPTemplates {
 
 export default function TemplateLoader({
   templates,
+  uriInfo,
+  dynamicLoader = React.lazy,
 }: {
+  uriInfo: UriInfo | undefined;
   templates: WPTemplates;
+  dynamicLoader: (loader: () => Promise<Template>) => React.ComponentType;
 }): JSX.Element | null {
-  const pageInfo = useNextUriInfo();
-  const Component = dynamic(() => resolveTemplate(pageInfo, templates));
+  const Component = dynamicLoader(() => resolveTemplate(uriInfo, templates));
```

Comment:
This could create an issue when `useUriInfo` initially returns `undefined`. Perhaps resolveTemplate should do nothing if `uriInfo` is undefined. It looks like currently it will return the index component.

Reply:
It looks like `useUriInfo` will return `undefined` only on the following conditional: https://github.com/wpengine/headless-framework/blob/canary/packages/headless/src/api/hooks.ts#L155-L159

Is the only time that it'd return undefined if there's a `uriPrefix` set and you're not viewing a page with the prefix?

Reply:
I wonder if throwing an exception in `resolveTemplate` if `pageInfo` is missing is a better option? 🤔

My thought there is pretty much everything depends on `pageInfo` as far as Data Fetchers go.

Reply:
Hmm, I would think it's possible for `useQuery` to return `undefined` data since it will be using `useEffect` and will have to return something initially and update state once the request is done. You might not hit this if using SSR/SSG and calling `useUriInfo` with the URI we use on the server side, but you would potentially hit it on the frontend.